liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateTableGenerator.java

```diff
@@ -279,7 +283,38 @@ public ValidationErrors validate(CreateTableStatement createTableStatement, Data
             buffer.append(",");
         }
 
+        /*
+         * In the current syntax, UNIQUE constraints can only be set per column on table creation.
+         * To alleviate this problem we combine the columns of unique constraints that have the same name.
+         */
+        HashMap<String, UniqueConstraint> uniqueConstraintMap = new HashMap<>();
+        List<UniqueConstraint> aggregatedUniqueConstraints = new LinkedList<>();
         for (UniqueConstraint uniqueConstraint : statement.getUniqueConstraints()) {
+            if (uniqueConstraint.getConstraintName() == null) {  // Only combine uniqueConstraints that have a name.
+                aggregatedUniqueConstraints.add(uniqueConstraint);}
+            else {
+                String constraintName = uniqueConstraint.getConstraintName();
+                if (uniqueConstraintMap.containsKey(constraintName)) { // if we have seen the uniqueContraint before ..
+                    UniqueConstraint oldUniqueConstraint = uniqueConstraintMap.get(constraintName);
+                    // check if it is really the same constraint
+                    if (oldUniqueConstraint.shouldValidateUnique() != uniqueConstraint.shouldValidateUnique()) {
+                        aggregatedUniqueConstraints.add(uniqueConstraint);  // if not, add it to the aggregated constraints
```

Comment:
Maybe it is better to throw an error here about conflicting column information?

Reply:
Maybe at least a warning. I'll add that in as I merge it